pyface/action/listening_action.py

```diff
@@ -102,32 +107,31 @@ def _enabled_name_changed(self, old, new):
         obj = self.object
         if obj is not None:
             if old:
-                obj.on_trait_change(self._enabled_update, old, remove=True)
+                obj.observe(self._enabled_update, old, remove=True)
             if new:
-                obj.on_trait_change(self._enabled_update, new)
-        self._enabled_update()
+                obj.observe(self._enabled_update, new)
+        self._enabled_update(event=None)
```

Comment:
let's update the signature of `_enabled_update` and `_visible_update` and remove this way of calling the method.

```suggestion
        self._enabled_update()
```